### DISCUSSION TITLE
Trying to batch crop images

### DISCUSSION THREAD

**Discussion by moutasem1989** [ImageMagick/ImageMagick#7059] [Help]
### ImageMagick version

7.1.1-28

### Operating system, version and so on

Debian 12

### Description

I have 100 pictures in several subfolders named "localmedia_*" in diffrent aspect reatios and sizes. I want to crop all of them in terminal to square without renaming, scaling, stretching or changing location. simply crop square to the center.
This command successfully returns a list of all the images:

`find . -type f -name "localmedia_*"`

similar to this:

> ./Contents/posters/localmedia_6273d593da3e3f94e404622d39219443ff491fdc
> ./Contents/posters/localmedia_13d13e34785396b31469331563de2fa298714e67
> ./Contents/posters/localmedia_44c9c62fb4976dd078bba17be271cb2cdb76e71a

note that image files do not have an extention. the images are also of diffrent sizes.

Applying a Mogrify command as following to one of the images results in an error:

`magick mogrify -gravity Center -crop 1:1 ./Contents/posters/localmedia_44c9c62fb4976dd078bba17be271cb2cdb76e71a`

> mogrify: geometry does not contain image ("60x60+0+30") ./Contents/posters/localmedia_44c9c62fb4976dd078bba17be271cb2cdb76e71a' @ warning/transform.c/CropImage/599.

can someone help me figure out what i am doing wrong ?

The goal is to crop all the images with a single command as following:

`find . -type f -name "localmedia_*" -exec magick mogrify -gravity Center -crop 1:1 {} \;`

**Comment by fmw42** [comment]
Mogrify was likely not upgraded to support the aspect ratio crop. It likely only exists currently in magick.
